crates/typst-cli/src/update.rs

```diff
@@ -0,0 +1,311 @@
+use std::{
+    env, fs,
+    io::{self, BufReader, Cursor, ErrorKind, Read, Seek},
+    path::Path,
+};
+
+use semver::Version;
+use serde::Deserialize;
+use typst::{diag::bail, diag::StrResult, eval::eco_format};
+use xz2::bufread::XzDecoder;
+use zip::ZipArchive;
+
+use crate::args::UpdateCommand;
+
+const TYPST_GITHUB_ORG: &str = "typst";
+const TYPST_REPO: &str = "typst";
+
+/// Self update the typst CLI binary.
+///
+/// Fetches a target release or the latest release (if no version was specified)
+/// from GitHub, unpacks it and self replaces the current binary with the
+/// pre-compiled asset from the downloaded release.
+pub fn update(command: UpdateCommand) -> StrResult<()> {
+    match self_update_permitted()? {
+        SelfUpdatePermission::Deny => {
+            eprintln!("Self-update is disabled for this build of the typst cli");
+            eprintln!("You should probably use your system package manager");
+            bail!("update failed");
```

Comment:
this does not seem accurate. `self_update_permitted` checks for write-access. self-update support is definitely in the binary if we reach this point.

Reply:
Removed, along with the enum and function.